src/object_store/src/object/mod.rs

```diff
@@ -891,11 +891,22 @@ pub async fn build_remote_object_store(
             set your endpoint to the environment variable RW_S3_ENDPOINT.");
             panic!("Passing s3-compatible is not supported, please modify the environment variable and pass in s3.");
         }
-        minio if minio.starts_with("minio://") => ObjectStoreImpl::S3(
-            S3ObjectStore::with_minio(minio, metrics.clone(), config.clone())
-                .await
-                .monitored(metrics, config),
-        ),
+        minio if minio.starts_with("minio://") => {
+            if env_var_is_false_or(RW_USE_OPENDAL_FOR_S3, false) {
```

Comment:
Nit: Why introduce `env_var_is_false_or` (in some last PR) when we already have `env_var_is_true_or`? IMO it can be very confusing to have some thing like `if some_is_false {} else {}` unless we need to do nothing for the `true` branch. Isn't clearer to write `if env_var_is_true_or(RW_USE_OPENDAL_FOR_S3, false) { Opendal } else { S3 }` here?

Reply:
I found it confusing too but I think the point here is to make the default value for `RW_USE_OPENDAL_FOR_S3` to be `true` when it is not set. A clearer way I can think of is to introduce a config `use_opendal_for_s3` with default value to be true and can be overridden by `RW_USE_OPENDAL_FOR_S3`.

Reply:
I guess the if-else structure is not quite related to the default value? No matter the default value is true or false, we can write `if env_var_is_true_or(RW_USE_OPENDAL_FOR_S3, default) { Opendal } else { S3 }`.